### PR TITLE
Add filter to select component

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | Add filter mode to `select-ripe`. Also fixes scroll bug. |
| Decisions | - |
| Animated GIF | imagem |

### CODE REVIEW

File: vue/components/ui/molecules/select/select.vue [hunk 1]
```diff
@@ -261,8 +301,7 @@ export const Select = {
         },
         scrollTo(index) {
             const dropdown = this.$refs.dropdown.$refs.dropdown;
-            const dropdownElements = document.getElementsByClassName("dropdown-item");
-
+            const dropdownElements = dropdown.getElementsByClassName("dropdown-item");
```
Comment (NFSS10) [inline vue/components/ui/molecules/select/select.vue]: Maybe use `$refs` here too

File: vue/components/ui/molecules/select/select.vue [hunk 2]
```diff
@@ -315,13 +360,21 @@ export const Select = {
...
+            this.highlight(0, scroll);
+        },
+        onPageUpKey(scroll = true) {
             this.openDropdown();
-            this.highlight(this.options.length - 1);
+            this.highlight(0, scroll);
+        },
+        onPageDownKey() {
+            this.openDropdown();
+            this.highlight(this.options.length - 1, true);
```
Comment (NFSS10) [inline vue/components/ui/molecules/select/select.vue]: These behaviours are the same, the only thing that changed was the keys. Create a method for scroll to the bottom/top and add it to theses onKeys methods.

File: vue/components/ui/molecules/select/select.vue [hunk 3]
```diff
@@ -401,6 +451,7 @@ export const Select = {
         classes() {
             const base = {};
             if (this.align) base[`select-align-${this.align}`] = this.align;
+            if (this.filter) base["select-filter"] = true;
```
Comment (joamag) [inline vue/components/ui/molecules/select/select.vue:454]: move this to the base object (it's a boolean thing ;))

File: vue/components/ui/molecules/select/select.vue [hunk 4]
```diff
@@ -390,6 +432,14 @@ export const Select = {
         }
     },
     computed: {
+        filteredOptions() {
+            if (!this.filter || !this.filterText) return this.options;
+            return this.options.filter(
+                option =>
+                    option.label &&
```
Comment (joamag) [inline vue/components/ui/molecules/select/select.vue:439]: make it possible to provide the function for filtering

File: vue/components/ui/molecules/select/select.vue [hunk 5]
```diff
@@ -236,6 +272,9 @@ export const Select = {
         openDropdown() {
             if (this.disabled || this.visibleData) return;
             this.visibleData = true;
+            if (!this.filter) return;
```
Comment (joamag) [inline vue/components/ui/molecules/select/select.vue:275]:
```javascript
if(this.filter) {
...
}
```

do not return immediately use block condition instead

**Comment by joamag** [comment]
@gcandal we'll need to back-port this stuff to white (it will take some time ;))

**Comment by joamag** [comment]
@gcandal I'm getting this error under some circumstances (typing fast and clicking on dropdown open)... 

Screenshot 2020-06-15 at 11 55 47

**Comment by joamag** [comment]
@gcandal let's take this chance to add some unit testing to this guy 🚀

**Comment by gcandal** [comment]
@3rdvision do you still remember what this is? :) Please grab this one 💪 

- Add unit tests
- Fix:
image

- Fix:
image

**Comment by joamag** [comment]
Closing this for now

**Pull request closed without merging**